tests/indexes_/test_atlas_indexes.py

```diff
@@ -0,0 +1,204 @@
+            index.name,
+            connection.introspection.get_constraints(
+                cursor=None,
+                table_name=model._meta.db_table,
+            ),
+        )
+        editor.remove_index(index=index, model=model)
+        self.assertNotIn(
+            index.name,
+            connection.introspection.get_constraints(
+                cursor=None,
+                table_name=model._meta.db_table,
+            ),
+        )
+
+    def test_simple(self):
+        with connection.schema_editor() as editor:
+            index = SearchIndex(
+                name="recent_article_idx",
+                fields=["number"],
+            )
+            editor.add_index(index=index, model=Article)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    def test_multiple_fields(self):
+        with connection.schema_editor() as editor:
+            index = SearchIndex(
+                name="recent_article_idx",
+                fields=["headline", "number", "body", "data", "embedded", "auto_now"],
+            )
+            editor.add_index(index=index, model=Article)
+            index_info = connection.introspection.get_constraints(
+                cursor=None,
+                table_name=Article._meta.db_table,
+            )
+            expected_options = {
+                "dynamic": False,
+                "fields": {
+                    "auto_now": {"type": "date"},
+                    "body": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "data": {"dynamic": False, "fields": {}, "type": "document"},
+                    "embedded": {"dynamic": False, "fields": {}, "type": "embeddedDocuments"},
+                    "headline": {
+                        "indexOptions": "offsets",
+                        "norms": "include",
+                        "store": True,
+                        "type": "string",
+                    },
+                    "number": {
+                        "indexDoubles": True,
+                        "indexIntegers": True,
+                        "representation": "double",
+                        "type": "number",
+                    },
+                },
+            }
+            self.assertCountEqual(index_info[index.name]["columns"], index.fields)
+            self.assertEqual(index_info[index.name]["options"], expected_options)
+            self.assertAddRemoveIndex(editor, Article, index)
+
+    def test_field_not_exists(self):
+        index = SearchIndex(
+            name="recent_article_idx",
+            fields=["headline", "non_existing_name"],
+        )
+        msg = "Article has no field named 'non_existing_name'"
+        with self.assertRaisesMessage(FieldDoesNotExist, msg), connection.schema_editor() as editor:
+            editor.add_index(index=index, model=Article)
```

Comment:
This test may be unnecessary since it's already validated by a system check:
```
search_indexes.Article: (models.E012) 'indexes' refers to the nonexistent field 'headline'.
```